src/slice.cr

```diff
@@ -534,6 +534,8 @@ struct Slice(T)
   end
 
   # See `Object#hash(hasher)`
+  # Be awary: `Slice(UInt8)` is hashed as binary data, and not as "an array of small
```

Comment:
I'm not sure how this is relevant, since the difference between these two options is impossible to tell from the outside, apart from speed. We don't tend to document optimizations.